### DISCUSSION TITLE
SocketIO SID

### DISCUSSION THREAD

**Discussion by SSC-Sheffield** [miguelgrinberg/Flask-SocketIO#1741] [Q&A]
When using the demo examples of creating a flask app server with socketio  I cannot seen to get an SID assigned. 
`app = Flask(__name__)`
`socketio = SocketIO(app, async_mode="gevent")`

`@socketio.event
def connect(sid):
    print(sid, 'connected')`

returns None Connected in the server terminal window. 

The JS client has 
`const socket = io('http://localhost:5000');`
`socket.on('connect', () => {
    console.log('connected');
    })`

which gets passed successfully to the console. 

I'm trying to read temperature readings from an ADC connected to a raspberry pi. There is some feedback from the client - ie what type of temperature sensor we want to read. The server then needs to find the correct data and return to the client. 

It's worked intermittently. 
Any help would be much appreciated!

**Comment by miguelgrinberg** [comment]
The connect handler does not take the `sid` as an argument. You should use `request.sid` inside the handler instead.
